.github/workflows/ci.yaml

```diff
@@ -27,12 +27,7 @@ jobs:
           python-version: ${{matrix.python}}
       - name: Install dependencies
         run: |
-          if test ${{matrix.python}} = 2.7; then
-            python -m pip install pyparsing==2.4.7 'PyYAML<6.0'
-          fi
-          python -m pip install --upgrade pip setuptools
-          python -m pip install mock pytest pytest-cov
-          python setup.py build develop
+          python -m pip install -U -e .[test] pytest-cov -c .github/workflows/constraints.txt
```

Comment:
Actually out of curiosity why isn't `pytest-cov` a test dependency?

Reply:
It isn't *actually* needed to run the tests locally, but is something we're seem to be interested in seeing in GitHub Actions.

Reply:
To further clarify, it isn't needed to run `pytest` on this repository unless you manually add `--cov` to the invocation. So I would deem it the caller's responsibility to install it if they'd like to use it.

Reply:
Part of me wants to advocate for an additional `extras_require.coverage` section but I'll hold my peace until the exact moment that there is more than a single coverage-specific dependency.